app/code/Magento/InventoryCatalog/Model/ResourceModel/AddIsInStockFieldToCollection.php

```diff
@@ -21,28 +22,47 @@ class AddIsInStockFieldToCollection
      */
     private $stockIndexTableProvider;
 
+    /**
+     * @var DefaultStockProviderInterface
+     */
+    private $defaultStockProvider;
+
     /**
      * @param StockIndexTableNameResolverInterface $stockIndexTableProvider
+     * @param DefaultStockProviderInterface $defaultStockProvider
```

Comment:
could you please add parameter in the Backward Compatible way (add nullification for new parameter)